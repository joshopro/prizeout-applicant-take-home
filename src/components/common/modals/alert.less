@import '../../../utils/styles/mixins.less';

.alert {
	@exclamation: ~'"\f06a"';
	@check: ~'"\f058"';
	@errorColor: var(--color-red-400);
	@warningColor: var(--color-yellow-400);
	@successColor: var(--color-green-400);
	
	.flexbox(@justifyContent: flex-start; @flexFlow: row nowrap);
	
	margin-bottom: var(--space-m);
	
	&--error {
		.alert-style(@exclamation; @errorColor);
	}
	
	&--warning {
		.alert-style(@exclamation; @warningColor);
	}
	
	&--success {
		.alert-style(@check; @successColor);
	}
	
	&__icon {
		margin-right: var(--space);
		
		&::before {
			display: inline-block;
			vertical-align: middle;
			font-weight: 300;
			font-family: var(--font-family-fa);
			font-size: 1rem;
			font-style: normal;
			font-variant: normal;
			text-rendering: auto;
			-webkit-font-smoothing: antialiased;
		}
	}
	
	&__icon,
	&__message {
		line-height: 1.2;
	}
	
	
	
	/*
		Local mixin to update icon and color based on alert type
	*/
	
	.alert-style(@iconType; @iconColor;) {
		.alert__icon {
			&::before {
				content: @iconType;
				color: @iconColor;
			}
		}
	}
}
