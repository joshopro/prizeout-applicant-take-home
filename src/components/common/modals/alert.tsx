import React from 'react';
import Classnames from 'classnames';
import PropTypes from 'prop-types';

import './alert.less';

interface AlertProps {
    message: string;
    type: 'error' | 'warning' | 'success' | undefined;
}

export const Alert: React.FC<AlertProps> = ({ message, type }): React.ReactElement => {
    const classes: string = Classnames(`alert`, `alert--${type}`);

    return (
        <div data-testid="alert-icon" className={classes} role="alert" aria-atomic="true">
            <span className="alert__icon"></span>
            <p className="alert__message">{message}</p>
        </div>
    );
};

Alert.propTypes = {
    message: PropTypes.string.isRequired,
    type: PropTypes.oneOf(['error', 'warning', 'success', undefined]),
};
