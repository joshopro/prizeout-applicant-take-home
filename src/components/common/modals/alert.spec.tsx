import { render, screen } from '../../../testing/test-utils';
import React from 'react';
import '@testing-library/jest-dom';

import { Alert } from './alert';

describe('Test Alert Component', () => {
    const message = `Alert message`;
    const type = `error`;

    // Matches snapshot
    test('Component matches snapshot', () => {
        const { asFragment } = render(<Alert message={message} type={type} />);

        expect(asFragment()).toMatchSnapshot();
    });

    // Alert text renders
    test('Alert text is displayed', () => {
        render(<Alert message={message} type={type} />);

        expect(screen.getByText(message)).toBeVisible();
    });

    // Alert icon renders based on alert type, check classes
    test('Alert icon display is based on alert type', () => {
        render(<Alert message={message} type={type} />);

        const alertId = screen.getByTestId('alert-icon');

        expect(alertId).toHaveClass(`alert--${type}`);
    });
});
