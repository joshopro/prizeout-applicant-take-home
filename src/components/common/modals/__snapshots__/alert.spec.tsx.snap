// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Alert Component Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-atomic="true"
    class="alert alert--error"
    data-testid="alert-icon"
    role="alert"
  >
    <span
      class="alert__icon"
    />
    <p
      class="alert__message"
    >
      Alert message
    </p>
  </div>
</DocumentFragment>
`;
